File: src/csharp/graphql/entitygraphql/Models/UserOrderStats.cs
namespace GraphqlEntityGraphQL.Models;

public class UserOrderStats
{
    public int UserId { get; set; }
    public string UserName { get; set; } = "";
    public int TotalOrders { get; set; }
    public double TotalValue { get; set; }
    public double AverageOrderValue { get; set; }
}

public class ComplexOrdersResult
{
    public int PeriodDays { get; set; }
    public int TotalUsers { get; set; }
    public List<UserOrderStats> Data { get; set; } = [];
}

File: src/kotlin/ktor/src/main/kotlin/com/benchmark/models/UserOrderStats.kt
package com.benchmark.models

import kotlinx.serialization.Serializable

// Mirrors UserOrderStats in contracts/grpc/benchmark.proto. The previous
// ComplexOrderResult carried {userId, email, order_count, total_amount,
// avg_amount, days_since_first_order} -- a shape no other implementation
// emitted. See contracts/rest/canonical-payloads.md.
@Serializable
data class UserOrderStats(
    val userId: Int,
    val userName: String,
    val totalOrders: Long,
    val totalValue: Double,
    val averageOrderValue: Double,
)
